src/Altinn.App.Core/Features/Options/AppOptionsFileHandler.cs

```diff
@@ -21,20 +26,20 @@ public AppOptionsFileHandler(IOptions<AppSettings> settings)
         }
 
         /// <inheritdoc/>
-        public async Task<List<AppOption>> ReadOptionsFromFileAsync(string optionId)
+        public async Task<List<AppOption>?> ReadOptionsFromFileAsync(string optionId)
         {
-            string legalPath = _settings.AppBasePath + _settings.OptionsFolder;
+            string legalPath = Path.Join(_settings.AppBasePath, _settings.OptionsFolder);
             string filename = legalPath + optionId + ".json";
             PathHelper.EnsureLegalPath(legalPath, filename);
 
             if (File.Exists(filename))
             {
                 string fileData = await File.ReadAllTextAsync(filename, Encoding.UTF8);
-                List<AppOption> options = JsonConvert.DeserializeObject<List<AppOption>>(fileData)!;
+                List<AppOption> options = JsonSerializer.Deserialize<List<AppOption>>(fileData, JSON_SERIALIZER_SETTINGS)!;
                 return options;
             }
 
-            return new List<AppOption>();
+            return null;
```

Comment:
I think we previously have said that for enumerable types we generally return an empty list instead of null. Any reason to change this?

Reply:
The most obvious reason is that the controller checks for NULL in order to return a 404 status, and that functionality is currently broken.

The other, more general is that there is a difference between `MY-OPTIONS` has no values, and that `MY-OPTIONS` does not exist. In other cases where there is no difference between "this instance has no data elements" and "the list of data elements is empty", the policy makes sense.

With nullable reference types the policy does not make much sense anymore in my opinion, but that is a different discussion.